lightwood/analysis/nc/icp.py

```diff
@@ -207,17 +209,18 @@ def predict(self, x, significance=None):
                 cal_scores = self.cal_scores[self.condition((x[j, :], c))][::-1]
                 n_cal = cal_scores.size
 
-                idx_left = np.searchsorted(cal_scores, nc, 'left')
-                idx_right = np.searchsorted(cal_scores, nc, 'right')
-                n_gt = n_cal - idx_right
-                n_eq = idx_right - idx_left + 1
-
-                p[j, i] = n_gt / (n_cal + 1)
+                n_eq = 0
+                n_gt = 0
+                for cal_score in cal_scores:
+                    if cal_score == nc:
+                        n_eq += 1
+                    elif nc < cal_score:
+                        n_gt += 1
 
                 if self.smoothing:
-                    p[j, i] += (n_eq * np.random.uniform(0, 1, 1)) / (n_cal + 1)
+                    p[j, i] = (n_gt + n_eq * random_uniform_dist) / (n_cal + 1)
                 else:
-                    p[j, i] += n_eq / (n_cal + 1)
+                    p[j, i] = (n_gt + n_eq) / (n_cal + 1)
```

Comment:
The math here has changed; previously you were adding to the cell and now you're replacing it. Why?

Reply:
because it's faster, we skip 1x assignment and turn 2x divisions into 1x divsion

Reply:
increases speed by removing 2x costly operations (1 assignment, 1 division)